common/adder/adder.xml

```diff
@@ -0,0 +1,103 @@
+<?xml version="1.0" encoding="utf-8"?>
+<component version="1.4" xmlns="http://schemas.circuit-diagram.org/circuitDiagramDocument/2012/component/xml">
+  <declaration>
+    <meta name="name" value="Adder" />
+    <meta name="minsize" value="50" />
+    <meta name="author" value="Corsaka" />
+    <meta name="guid" value="8f764035-18c6-4421-ac7f-5cbd327d4e78" />
+
+    <property name="Type" type="enum" default="Full" serialize="t" display="Type">
+      <option>Full</option>
+      <option>Half</option>
+    </property>
+
+    <configurations>
+      <configuration name="Full Adder" value="Type:Full" implements="full" />
+      <configuration name="Half Adder" value="Type:Half" implements="half" />
+    </configurations>
+  </declaration>
+  <connections>
+    <group conditions="horizontal">
+      <connection name="A" start="_Start-7x-30y" end="_Start-7x-60y" edge="End" />
+      <connection name="B" start="_End+7x-30y" end="_End+7x-60y" edge="End" />
+      <connection name="Sum" start="_Middle+30y" end="_Middle+60y" edge="End" />
+      <connection name="Cin" start="_Start-15x" end="_Start-45x" edge="End" />
+      <connection name="Cout" start="_End+15x" end="_End+45x" edge="End" />
+    </group>
+    <group conditions="!horizontal">
+      <connection name="A" start="_Start-35y" end="_Start-30x-35y" edge="End" />
+      <connection name="B" start="_Start+35y" end="_Start-30x+35y" edge="End" />
+      <connection name="Sum" start="_End" end="_End+30x" edge="End" />
+      <connection name="Cin" start="_Middle-45y" end="_Middle-75y" edge="End" />
+      <connection name="Cout" start="_Middle+45y" end="_Middle+75y" edge="End" />
+    </group>
+  </connections>
+  <render>
+    <group conditions="horizontal">
+      <!-- draw shape -->
+      <line start="_Start+30y" end="_Start-30x-30y" />
+      <line start="_Start+30y" end="_End+30y" />
+      <line start="_Start-30x-30y" end="_Start+15x-30y" />
+      <line start="_Start+15x-30y" end="_Middle" />
+      <line start="_Middle" end="_End-15x-30y" />
+      <line start="_End-15x-30y" end="_End+30x-30y" />
+      <line start="_End+30y" end="_End+30x-30y" />
+      <!-- draw connection lines -->
+      <line start="_Start-7x-30y" end="_Start-7x-60y" />
```

Comment:
The Cin line should begin at `_Start` and end at `_Middle-{something}` and similar for Cout. This way the component will resize properly by extending the start and end lines.

At the moment the resize points (shown as blue circles below) don't appear in the right places:

![image](https://user-images.githubusercontent.com/5303088/75492550-e7b88580-59af-11ea-998d-1ff3da38d520.png)

Reply:
I don't believe the component will fit if I do that; How would I increase the distance between `_Start` and `_End`?

Reply:
You can increase the minimum distance by changing this:

```xml
<meta name="minsize" value="60" />
```